api/tests/Behat/bootstrap/v2/DeputyManagement/DeputyManagementTrait.php

```diff
@@ -0,0 +1,174 @@
+     */
+    public function iShouldBeAbleToUpdateFirstnameLastnamePostcode()
+    {
+        $this->iVisitAdminEditUserPageForInteractingWithUser();
+
+        $this->assertElementOnPage('input[name="admin[firstname]"]');
+        $this->assertElementOnPage('input[name="admin[lastname]"]');
+        $this->assertElementOnPage('input[name="admin[addressPostcode]"]');
+
+        $this->completedFormFields['admin[firstname]'] = $this->faker->firstname;
+        $this->completedFormFields['admin[lastname]'] = $this->faker->lastname;
+        $this->completedFormFields['admin[addressPostcode]'] = $this->faker->postcode;
+    }
+
+    /**
+     * @Then I should not be able to update the deputies email address
+     */
+    public function iShouldNotBeAbleToUpdateEmail()
+    {
+        $this->iVisitAdminEditUserPageForInteractingWithUser();
+        $this->assertElementNotOnPage('input[name="admin[email]"]');
+    }
+
+    /**
+     * @When I update the details of the deputy available to me
+     */
+    public function iUpdateTheUsersDetailsAvailableToMe()
+    {
+        foreach ($this->completedFormFields as $fieldName => $fieldValue) {
+            $this->fillField($fieldName, $fieldValue);
+        }
+
+        $this->pressButton('Update user');
+        $this->em->clear();
+    }
+
+    /**
+     * @Then the deputies details should be updated
+     */
+    public function theDeputyShouldBeUpdated()
+    {
+        $this->assertOnAlertMessage('Your changes were saved');
+        $id = $this->interactingWithUserDetails->getUserId();
+        $user = $this->em->getRepository(User::class)->find($id);
+
+        $comparisonSubjectMessage = sprintf('Queried DB for User with id %s against form values entered', $id);
+        $this->assertStringEqualsString(
+            $this->completedFormFields['admin[firstname]'],
+            $user->getFirstname(),
+            $comparisonSubjectMessage
+        );
+        $this->assertStringEqualsString(
+            $this->completedFormFields['admin[lastname]'],
+            $user->getLastname(),
+            $comparisonSubjectMessage
+        );
+        $this->assertStringEqualsString(
+            $this->completedFormFields['admin[addressPostcode]'],
+            $user->getAddressPostcode(),
+            $comparisonSubjectMessage
+        );
+
+        if ($this->emailShouldBeUpdated($this->loggedInUserDetails->getUserRole(), $user->getRoleName())) {
+            $this->assertStringEqualsString(
+                $this->completedFormFields['admin[email]'],
+                $user->getEmail(),
+                $comparisonSubjectMessage
+            );
+        }
+
+        $this->interactingWithUserDetails = null;
+        $this->completedFormFields = [];
+    }
+
+    private function emailShouldBeUpdated(string $loggedInUserRole, string $userToBeUpdatedRole): bool
+    {
+        $rolesAdminManagersCanUpdateEmail = [
+            User::ROLE_LAY_DEPUTY,
```

Comment:
Is this weird?

Reply:
As in the case of it compared to the others? :)

Reply:
Oh wow not sure how I managed that - it's the other roles that are wrong here so I'll update them.